app/interactors/fetch_oauth_user.rb

```diff
@@ -0,0 +1,22 @@
+class FetchOauthUser
+  attr_reader :auth
+  private :auth
+
+  def initialize(auth)
+    @auth = auth
+  end
+
+  def call
+    find_user_by_email || find_social_profile_user
```

Comment:
I guess, we should try to find social profile, before finding by email.

If user has already signed with this social account, then he should have Social Profile, so it's more reasonable to look for it, and if we haven't found one - find by email, and create new social profile for him